.github/workflows/_runner_determinator_script_sync.yaml

```diff
@@ -0,0 +1,42 @@
+name: runner-determinator
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [main]
+    paths:
+      - .github/workflows/_runner-determinator.yaml
+      - .github/workflows/_runner_determinator_script_sync.yaml
+      - .github/workflows/scripts/runner_determinator.py
+
+jobs:
+  python-script-sync-check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            .github
+
+      - name: Extract the script from runner_determinator
+        run: |
+            # Runner determinator files
+            RUNNER_DETERMINATOR_WORKFLOW_FILE=.github/workflows/_runner-determinator.yml
+            RUNNER_DETERMINATOR_PYTHON_SCRIPT_FILE=.github/scripts/runner_determinator.py
+
+            # Parse the job file, extract the script and run it, up to the final EOF,
+            # to generate the python file in the local folder
+            yq '.jobs.runner-determinator.steps[] | select(.id == "hardcode-script") | .run' \
+                "${RUNNER_DETERMINATOR_WORKFLOW_FILE}" | sed '/^EOF$/q' | bash
+
+            set +e
+            DIFF="$(diff "$(basename ${RUNNER_DETERMINATOR_PYTHON_SCRIPT_FILE})" ${RUNNER_DETERMINATOR_PYTHON_SCRIPT_FILE})"
```

Comment:
Does this ignore that any trailing new-lines at the end of either copy of the script? (would be nice if it does!)

Reply:
`diff -B` would ignore blank line in any position within the script - @ZainRizvi would you like me to add the `-B` flag? 

Alternatively, I could parse both files with some obscure `sed` command like:

```
# Delete all trailing blank lines at end of file (only).
sed -e :a -e '/^\n*$/{$d;N;};/\n$/ba' file
```